nornir_nautobot/utils/helpers.py

```diff
@@ -31,3 +32,9 @@ def import_string(dotted_path):
         return getattr(importlib.import_module(module_name), class_name)
     except (ModuleNotFoundError, AttributeError):
         return None
+
+
+def get_stack_trace(exc: Exception) -> str:
+    """Converts the provided exception's stack trace into a string."""
+    stack_trace_lines = traceback.format_exception(type(exc), exc, exc.__traceback__)
+    return "\n".join(stack_trace_lines)
```

Comment:
Do we need the `\n` here? I only ask since the screenshot showed what seemed like extra new lines.

Reply:
we do need it here or it just comes out as a single line,  but I think I might see where the extra liens are coming from - looking into it

<img width="1604" alt="Screenshot 2024-03-21 at 11 49 58 AM" src="https://github.com/nautobot/nornir-nautobot/assets/47164813/86b3f020-5438-4720-be50-134c665685e1">

Reply:
Here's the traceback copied from the log for reference,  I do have an extra space at the bottom but between the frames there is only a single space.
```
Traceback (most recent call last):

  File "/usr/local/lib/python3.11/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/nornir_jinja2/plugins/tasks/template_file.py", line 43, in template_file
    text = t.render(host=task.host, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()

  File "/usr/local/lib/python3.11/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)

  File "/opt/nautobot/git/access_config_templates/calix_axos.j2", line 3, in top-level template code
    {% include './calix_axos/aaa.j2' %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/opt/nautobot/git/access_config_templates/calix_axos/aaa.j2", line 4, in top-level template code
    <authentication-order>{{ config_context["aaa"]["authentication-order"] }}</authentication-order>
    ^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/jinja2/sandbox.py", line 303, in getitem
    return obj[argument]
           ~~~^^^^^^^^^^

jinja2.exceptions.UndefinedError: 'collections.OrderedDict object' has no attribute 'aaa'

```

Reply:
Looking into this further there is no space between frames in the actual stack trace on the worker,  the `/n` wasn't actually needed in the join  - good eye @itdependsnetworks 
<img width="1009" alt="Screenshot 2024-03-21 at 12 21 53 PM" src="https://github.com/nautobot/nornir-nautobot/assets/47164813/a4761797-7eb3-4c20-a15c-58f4495dd01e">